commons/all/src/main/java/org/infinispan/commons/util/ByteQuantity.java

```diff
@@ -0,0 +1,82 @@
+package org.infinispan.commons.util;
+
+import static java.math.BigDecimal.TEN;
+
+import java.math.BigDecimal;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import org.infinispan.commons.logging.Log;
+import org.infinispan.commons.logging.LogFactory;
+
+/**
+ * Parser human-readable quantity of bytes.
+ *
+ * @since 11.0
+ */
+public final class ByteQuantity {
```

Comment:
We have the `MemoryUnit` class that already does a lot of this. Maybe we can consolidate?

Reply:
I have completely missed that 👍